### ISSUE TITLE
OMP threads and Cube Histogram

### ISSUE THREAD

**Issue by ajm-ska** [CARTAvis/carta-backend#642]
Edit from @veggiesaurus: This seems to occur on all platforms, for channel or cube histograms!

We have discovered an issue with the Cube Histogram calculation on MacOS where it can produce a slightly different result than on Linux (RedHat7 and Ubuntu).

A workaround is to define the environment variable `OMP_NUM_THREADS` in the shell on MacOS before starting the carta_backend. However, it only works properly when it is set in a factor of two sequence; 2, 4, 8, 16, 32, 64 etc. 
If it is set as 6, 12, or 50, for example, it exhibits the same issue.

Setting `OMP_NUM_THREADS` on Linux is apparently not needed.

The carta_backend flag `-omp_threads` has no noticeable effect.

The Cube Histogram calculation should give identical results on MacOS and Linux without needing OMP_NUM_THREADS defined in advance.

![Screen Shot 2020-10-14 at 8 38 44 PM](https://user-images.githubusercontent.com/20802551/95989930-7acdff00-0e5d-11eb-8a0e-94ba32ecc1a1.png)

**Comment by veggiesaurus** [comment]
@Jordatious saw a similar issue, where the FITS histogram was under-counting. Wonder if it's related.

**Comment by Jordatious** [comment]
Here's an example
![image](https://user-images.githubusercontent.com/10542136/96121756-c8e01280-0ef0-11eb-8828-d10840738b1d.png)
![image](https://user-images.githubusercontent.com/10542136/96121775-cf6e8a00-0ef0-11eb-9378-55e3b729f48b.png)
The peak of the histogram for the HDF5 version always seems to be higher.

**Comment by veggiesaurus** [comment]
I guess these are two separate issues:
1) The histogram results being inconsistent
2) The `omp_threads` flag not working in MacOS (this seems less critical. @ajm-asiaa can you make a separate issue for this if it is essential?)

**Comment by veggiesaurus** [comment]
branch `angus/test_histograms` has a simple unit test demonstrating this issue. The test fails whenever the number of threads is not a power of 2, with a histogram of uniform data between 0 and 1.

**Comment by veggiesaurus** [comment]
> Here's an example
> The peak of the histogram for the HDF5 version always seems to be higher.

@Jordatiou I'll deploy the fix on carta-testing soon in order for you to verify it's working properly.
